commit 5ad3244df44ceeab03dd9fdf43344d59ca00b4c6
Author: Sage Weil <sweil@redhat.com>
Date:   Fri Feb 24 22:25:39 2017 -0500

    mgr/DaemonServer: use process_pg_map_command helper for most things
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index 7d2e2abb2af..85eed18f55c 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -328,7 +328,19 @@ bool DaemonServer::handle_command(MCommand *m)
 #endif
     f.close_section();	// command_descriptions
     goto out;
-  } else {
+  }
+  else {
+    cluster_state.with_pgmap(
+      [&](const PGMap& pg_map) {
+	cluster_state.with_osdmap([&](const OSDMap& osdmap) {
+	    r = process_pg_map_command(prefix, cmdmap, pg_map, osdmap,
+				       f.get(), &ss, &odata);
+	  });
+      });
+  }
+
+  // fall back to registered python handlers
+  if (r == -EOPNOTSUPP) {
     // Let's find you a handler!
     MgrPyModule *handler = nullptr;
     auto py_commands = py_modules.get_commands();
